boards/arm/nucleo_g431rb/nucleo_g431rb.dts

```diff
@@ -59,7 +59,7 @@
 	status = "okay";
 };
 
-&clk_lse {
+&clk_hsi48 {
```

Comment:
ok, but why removing LSE ?
Are you sure it isn't required by other peripheral? 
At least it would be needed to understand why it was enabled in first place and that it is safe to disable it.

Reply:
> ok, but why removing LSE ?

When `clk_lse` is left as is - the clock provided to RNG is too slow. I've looked around and STM32F uses the `clk_hsi48` in
conjunction to `RNG` generator (which I do need for LAN865x (T1S) ethernet support).

> Are you sure it isn't required by other peripheral?

I did not observed (at least in my use case) any regressions

> At least it would be needed to understand why it was enabled in first place and that it is safe to disable it.

My understanding is that - as `nucleo_g491re` doesn't have ETH, it was enough to use low-speed external oscillator (`LSE`). However, maybe some people from STM ( @ABOSTM  or @FRASTM ) could provide more in-depth explanation.

Reply:
As I've said above - I just found the issue and fixed it in a way which I've found for similar STM32 board.

I would be more than happy to get feedback or insights from Zephyr community (especially from STM people)

Reply:
Just FYI, I'm STM32 maintainer (and also ST employee but since this is an open source project you should rather refer to maintainership status than the origin company). So please take into account my comment, I'm just trying to help you.

Reply:
@erwango - I see. 

The problem with this nucleo board is that ETH is not present on it. As a result the RNG is not required.

After adding T1S chip the RNG is required and current clock setup is too low (entrophy driver IRQ storm).

The fix was to use internal? 48 MHz source (`clk_hsi48`).  I don't know if this is the best solution - for sure `clk_lse` is too slow, so we need to decide how to change it.

I would appreciate any hints and comments on how to fix this issue in the correct way. That is why I've asked for some assistance from STM32 as this is their "development" board.

Reply:
> The problem with this nucleo board is that ETH is not present on it. As a result the RNG is not required.

It could be required for other purposes. RNG IP is available on most G4 (if not all) G4 SoCs while there is no ethernet at all on this series.

> After adding T1S chip the RNG is required and current clock setup is too low (entrophy driver IRQ storm).

This part I understand

> The fix was to use internal? 48 MHz source (clk_hsi48). I don't know if this is the best solution - for sure clk_lse is too slow, so we need to decide how to change it.

`clk_lse` could be used for other  purposes than RNG. You should not assume that it was there only for RNG.

> I would appreciate any hints and comments on how to fix this issue in the correct way. That is why I've asked for some assistance from STM32 as this is their "development" board.

- Add  `clk_hsi48` as you've done
- Don't remove `clk_lse`
- Add the following:
```
&rng {
	clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x04000000>,
		 <&rcc STM32_SRC_HSI48 CLK48_SEL(0)>;
};
```
(After check in the ref man, this last step is optional as this is the default CLK48SEL value).

At least this is the way it should be done on latest Zephyr. It could be different on earlier versions.